### ISSUE TITLE
Invalid value input is not handled properly

### ISSUE THREAD

**Issue by n0ruSh** [Qix-/color#239]
# How to reproduce:

1. Open [Sandbox](https://codesandbox.io/s/invalid-color-wg220?file=/src/index.js)
2. Check the console output

# Current behavior

* For invalid rgb input (E.g. "rgb(277,0,0)"), r is rounded to the nearest valid value `255` and a value is returned.
* For invalid hex input (E.g. "#p10000"), an exception is thrown.

# Expected behavior

Exceptions are expected to be thrown for all invalid inputs (i.e. for input that can't be turned into an equivalent color).

# Extra info

Probablly related to this code: https://github.com/Qix-/color-string/blob/4daceef89391f28eba415d043bbd1163c8f19a8c/index.js#L128

**Comment by Qix-** [comment]
I agree though this is a breaking change. It'd have to wait for v5 unfortunately.

**Comment by n0ruSh** [comment]
Any plan for v5 😃
